### DISCUSSION TITLE
TreeTable data里面必须传入children不能根据parentId自动识别为子节点根节点吗

### DISCUSSION THREAD

**Discussion by MineMineLXH** [layui/layui#1737] [问答]
### 议题条件

- [X] 我确认已查看官方使用文档：https://layui.dev ，但没有找到相关解决方案。
- [X] 我确认已在 [Issues](https://github.com/layui/layui/issues?q=is%3Aissue ) 中搜索过类似的问题，但没有找到相关解决方案。
- [X] 我已仔细阅读： 🍀[ Layui Issue 贡献指南](https://github.com/layui/layui/blob/main/CONTRIBUTING.md)

### 议题类型

功能请求

### 使用版本

V2.9.2

### 问题描述

现在treetable后台返回的json中包含了parentId，isParent，Children属性，不能只根据parentId直接自动识别父子节点展示吗？还有文档的异步加载子节点数据不太理解，如果启用异步加载子节点，父节点的url处返回的数据就只包含父节点数据吗？

### 业务代码

```auto
var inst=treeTable.render( {
    elem: '#ID-treeTable-demo',
    url: '/Sys/SysMenu/TreeMenuJson', // 此处为静态模拟数据，实际使用时需换成真实接口
    tree: {
        
        
    },
    toolbar: '#TPL-treeTable-demo',
    cols: [ [
        { type: 'checkbox', fixed: 'left' },
        { field: 'id', title: 'ID' },
        { field: 'title', title: '菜单名' },
        { field: 'icon', title: '图标' },
        { field: 'type', title: '类型' },
        { field: 'openType', title: '打开方式' },
        { field: 'href', title: '路由' },
        { field: 'IsEnable', title: '是否启用' },
        { fixed: "right", title: "操作", width: 190, align: "center", toolbar: "#TPL-treeTable-demo-tools" }
    ] ],
    page: true
} );
```

### 浏览器

chrome  122.0.6261.71

### 友好承诺

- [X] 我承诺将本着相互尊重、理解和友善的态度进行交流，共同维护 Layui 良好的社区氛围。

**Comment by Sight-wcg** [comment]
不能只根据parentId直接自动识别父子节点展示吗？

> 开启 isSimpleData
![image](https://github.com/layui/layui/assets/26325820/83e3c249-eb12-4ae4-b4a4-ffb5fab83bb8)

> 还有文档的异步加载子节点数据不太理解，如果启用异步加载子节点，父节点的url处返回的数据就只包含父节点数据吗？

是的

**Comment by MineMineLXH** [comment]
文档还是欠缺详细的说明啊，文档中并没有写明简单数据模式是什么样的json结构，个人使用反馈感觉layui原生的treetable较为复杂，之前使用的第三方插件的treetable，基本可以看一下代码直接上手，这个使用起来特别不方便
